File: couch/java/create/couch_create.java
// --------------------------------------------------------------
/*
	couch/java/create/couch_create.java

					Jun/23/2014

*/
// --------------------------------------------------------------
import	java.util.HashMap;
import	java.util.Set;

import  net.arnx.jsonic.JSON;
// --------------------------------------------------------------
public class couch_create
{

// --------------------------------------------------------------
public static void main(String[] args) throws Exception
{
	System.out.println("*** 開始 ***");

	HashMap <String, HashMap <String,String>>
		dict_aa = data_prepare_proc ();

	text_manipulate.dict_display_proc (dict_aa);

	final String url_collection = "http://localhost:5984/nagano";

	System.out.println("*** check bbb ***");

	get_uri.rest_delete_proc (url_collection);
	get_uri.rest_put_proc (url_collection,"{}","text/json");

	Set set_aaa = dict_aa.keySet ();

	for (Object key_aa: set_aaa)
		{
		String key = (String)key_aa;

		HashMap <String,String> unit_aa = dict_aa.get (key);

		String str_json_new = JSON.encode (unit_aa);
		String url_target = url_collection + "/" + key;
		get_uri.rest_put_proc (url_target,str_json_new,"text/json");
		}

	System.out.println("*** 終了 ***");
}
 
// --------------------------------------------------------------
static HashMap <String, HashMap<String,String>> data_prepare_proc ()
{
	HashMap <String, HashMap<String,String>> dict_aa
		= new HashMap <String, HashMap<String,String>> ();

	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2021","長野",91426,"2008-2-18");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2022","松本",35713,"2008-8-24");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2023","上田",49287,"2008-4-7");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2024","小諸",19256,"2008-3-21");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2025","岡谷",67392,"2008-5-25");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2026","塩尻",85647,"2008-9-9");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2027","茅野",67219,"2008-12-11");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2028","飯田",63981,"2008-5-27");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2029","中野",21548,"2008-1-5");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2030","諏訪",53297,"2008-7-4");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2031","駒ヶ根",39658,"2008-9-27");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2032","佐久",14256,"2008-10-24");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2033","伊那",31547,"2008-12-21");
	dict_aa = text_manipulate.dict_append_proc
		(dict_aa,"t2034","千曲",47251,"2008-11-29");

	return	dict_aa;
}

// --------------------------------------------------------------
}
// --------------------------------------------------------------

File: couch/scala/create/couch_create.scala
// -----------------------------------------------------------------
//	couch_create.scala
//
//					Jun/23/2014
//
// -----------------------------------------------------------------
import scala.collection.mutable
import scala.util.parsing.json.JSONObject

// -----------------------------------------------------------------
object couch_create
{
def main (args: Array[String])
{
	println	("*** 開始 ***")

	val url_collection = "http://localhost:5984/nagano"

	get_uri.rest_delete_proc (url_collection)
	get_uri.rest_put_proc (url_collection,"{}","text/json")

	val dict_aa = data_prepare_proc ()

	for (pair <- dict_aa)
		{
		val key = pair._1
		val unit_aa = pair._2.asInstanceOf [mutable.Map[String,String]]
		val unit_bb = unit_aa.toMap
		val obj = JSONObject (unit_bb)
		val json_str_new = obj.toString ()
//		println (json_str_new)
		val url_target = url_collection + "/" + key 
		get_uri.rest_put_proc (url_target,json_str_new,"text/json")
		}

	println	("*** 終了 ***")

}
// -----------------------------------------------------------------
def data_prepare_proc ():(mutable.Map[String,Object]) = {

	var dict_aa = mutable.Map[String,Object] ()

	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2021","長野",32945,"1998-8-5")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2022","松本",67381,"1998-6-21")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2023","上田",86173,"1998-5-12")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2024","小諸",73542,"1998-4-23")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2025","岡谷",41895,"1998-2-17")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2026","塩尻",69423,"1998-1-24")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2027","茅野",32897,"1998-3-21")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2028","飯田",85128,"1998-9-8")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2029","中野",27315,"1998-10-14")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2030","諏訪",35921,"1998-5-8")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2031","駒ヶ根",89276,"1998-7-3")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2032","佐久",42367,"1998-9-9")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2033","伊那",87192,"1998-2-12")
	dict_aa = text_manipulate.dict_append_proc (dict_aa,"t2034","千曲",78916,"1998-4-25")

	dict_aa
}
// -----------------------------------------------------------------
}
// -----------------------------------------------------------------
